<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>999</width>
    <height>658</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_v1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_k">
          <item>
           <widget class="QLabel" name="k_label">
            <property name="text">
             <string>K:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="k_spinBox">
            <property name="value">
             <number>3</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_speed">
          <item>
           <widget class="QLabel" name="speed_label">
            <property name="text">
             <string>Speed(ms): </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="speed_spinBox">
            <property name="value">
             <number>99</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_sample">
          <item>
           <widget class="QLabel" name="sample_label">
            <property name="text">
             <string>Sample per Cluster:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="sample_spinBox">
            <property name="value">
             <number>30</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_v3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_v2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_dim">
          <item>
           <widget class="QLabel" name="dim_label">
            <property name="text">
             <string>Dimension(2 or 3):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="dim_comboBox">
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>3</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_dist_m">
          <item>
           <widget class="QLabel" name="dist_m_label">
            <property name="text">
             <string>Distance method:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="dist_m_comboBox">
            <item>
             <property name="text">
              <string>l1_norm</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>l2_norm</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>l-infinity-norm</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_cent_m">
          <item>
           <widget class="QLabel" name="cent_m_label">
            <property name="text">
             <string>Centroid initializing method:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cent_m_comboBox">
            <item>
             <property name="text">
              <string>random_real</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>random_sample</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>k-means++</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_v2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_v3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_point_size">
          <item>
           <widget class="QLabel" name="point_size_label">
            <property name="text">
             <string>Point Size:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="point_size_doubleSpinBox">
            <property name="value">
             <double>3.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_centroid_size">
          <item>
           <widget class="QLabel" name="cent_size_label">
            <property name="text">
             <string>Centroid Size:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="cent_size_doubleSpinBox">
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_v1">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_v4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_file_name">
          <item>
           <widget class="QLabel" name="file_name_label">
            <property name="text">
             <string>File Name:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="file_name_comboBox">
            <item>
             <property name="text">
              <string>a.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>b.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>c.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>d.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>d_ring.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>e_corners_1.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>f_corners_2.txt</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>g_overlapping.txt</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="if_file_checkBox">
          <property name="text">
           <string>Data from File</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="generate_pushButton">
          <property name="text">
           <string>Generate Data</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_h">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="ViewWidget" name="viewWidget"/>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_iter">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_iter">
          <item>
           <widget class="QCheckBox" name="each_step_checkBox">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Go Each Step</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="iter_label">
            <property name="text">
             <string>Number of iterations:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="iter_spinBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="prev_pushButton">
            <property name="text">
             <string>Previous</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="next_pushButton">
            <property name="text">
             <string>Next</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="zoomOut_pushButton">
            <property name="text">
             <string>Zoom Out</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="zoomIn_pushButton">
            <property name="text">
             <string>Zoom In</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="zoomResize_pushButton">
            <property name="text">
             <string>Resize</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="turn_left_pushButton">
            <property name="text">
             <string>&lt;&lt;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="turn_right_pushButton">
            <property name="text">
             <string>&gt;&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="left_pushButton">
          <property name="text">
           <string>Left</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="up_pushButton">
            <property name="text">
             <string>Up</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="center_pushButton">
            <property name="text">
             <string>Center</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="down_pushButton">
            <property name="text">
             <string>Down</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="right_pushButton">
          <property name="text">
           <string>Right</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>999</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <addaction name="menuView"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionActual_Size">
   <property name="text">
    <string>Actual Size</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ViewWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>ViewWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>zoomIn_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>583</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomOut_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomResize_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomActualSize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>908</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>turn_left_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>turnLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>623</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>turn_right_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>turnRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>867</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>center_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>goCenter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>529</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>down_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>goDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>584</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>goLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>724</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>goRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>937</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>up_pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>goUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>474</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>zoomActualSize()</slot>
  <slot>zoomIn()</slot>
  <slot>zoomOut()</slot>
  <slot>turnLeft()</slot>
  <slot>turnRight()</slot>
  <slot>goUp()</slot>
  <slot>goDown()</slot>
  <slot>goLeft()</slot>
  <slot>goRight()</slot>
  <slot>goCenter()</slot>
 </slots>
</ui>
